test/NuGet.Core.Tests/NuGet.Commands.Test/RestoreResultTests.cs

```diff
@@ -327,6 +327,64 @@ public async Task RestoreResult_Commit_WritesDependencyGraphSpec()
             }
         }
 
+        [Theory]
+        [InlineData(true)]
+        [InlineData(false)]
+        public async Task RestoreResult_Commit_WithExistingDGSpecAndDidDGHashChange_WritesDependencyGraphSpec(bool didDGHashChange)
```

Comment:
nit: It'd be good to get away from repeating test class names in test method names.
non-nit: The criteria isn't clear here. I read the assertions as it should only write when the DGHash changed, because it already existed. Can the test name be clearer?

```suggestion
        public async Task CommitAsync_WithExistingDGSpec_OnlyWritesWhen_DidDGHashChangeIsTrue(bool didDGHashChange)
```

Reply:
The `_` needs to be `What_UnderWhatConditions_Expectations`. 

I merged your suggestion to what I had.